src/main/java/com/crowd/entity/Project.java

```diff
@@ -146,6 +146,29 @@ public void setDonateList(List<Donate> donateList) {
         this.donateList = donateList;
     }
 
+    public List<Donate> getApproveDonateList() {
+        List<Donate> approvedDonates = new ArrayList<>();
+        for (Donate donate : donateList) {
+            if (donate.isApproved()) {
+                approvedDonates.add(donate);
+            }
+        }
+        return approvedDonates;
+    }
+
+    public double getApprovedAmount() {
+        return getApproveDonateList().stream().mapToDouble(Donate::getAmount).sum();
+    }
+
+    public double percentApprovedDonate() {
+        double percent = getApprovedAmount() / needAmount * 100;
+        return NumberFormatter.format(percent);
```

Comment:
functions named `format` usually return string. Seems you should chose another name for your function.